vcr/vcr.go

```diff
@@ -303,73 +298,6 @@ func (c *vcr) Resolve(ID ssi.URI, resolveTime *time.Time) (*vc.VerifiableCredent
 	return &credential, nil
 }
 
-// Validate validates the provided credential.
-// The function returns nil when the credential is considered valid, the validation error otherwise.
-//
-// It accepts a few extra flags which configure the validation process:
-// * If the allowUntrusted bool is set to true, the credential is not checked for trust.
-//   This means that the validity does not depend on if the issuer-type combination is set to be trusted on this node.
-// * If the checkSignature is set to false, the signature will not be checked.
-//   If it is set to true, the signature must compute. Also, the used signing key must be valid at the validAt time.
-//   A signing key is considered valid if the issuer AND at least one (if any) of its controllers was active at the validAt time.
-// * If the validAt is not provided, validAt will be set to the current time.
-//
-// In addition to the signing key time checks, the following checks will be performed:
-// * The ID fields must be set
-// * The credential is not revoked (note: the revocation state is currently time independent)
-// * The type must contain exactly one type in addition to the default `VerifiableCredential` type.
-// * The issuanceDate must be before the validAt.
-// * The expirationDate must be after the validAt.
-func (c *vcr) Validate(credential vc.VerifiableCredential, allowUntrusted bool, checkSignature bool, validAt *time.Time) error {
-	if credential.ID == nil {
-		return errors.New("verifying a credential requires it to have a valid ID")
-	}
-
-	if validAt == nil {
-		now := timeFunc()
-		validAt = &now
-	}
-
-	// TODO: remove after great reset
-	revoked, err := c.isRevoked(*credential.ID)
-	if revoked {
-		return types.ErrRevoked
-	}
-	if err != nil {
-		return err
-	}
-
-	if checkSignature {
-		// check if the issuer was valid at the given time. (not deactivated, or deactivated controller)
```

Comment:
The verifier does not check for deactivated controllers of the signing DID document.